.github/workflows/main.yml

```diff
@@ -139,7 +139,7 @@ jobs:
           install\${{ env.PRESET }}\bin\power_grid_model_package_test; if(!$?) { Exit $LASTEXITCODE }
 
   build-cpp-test-macos:
-    runs-on: macos-13
+    runs-on: macos-14
```

Comment:
@mgovers, is the Python hack for 3.12 for brew still needed?

Reply:
I have removed it and it works!

Reply:
https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md#language-and-runtime shows that it runs python 3.12.2, so that's great!